accounts/accounts.go

```
@@ -46,41 +47,67 @@ func NewAccount(username string, hostname string) *Account {
 		Directory: directory,
 		Passwd:    "generated password",
 		NextId:    1,
+		Token:     token,
 	}
 }
 
-// TODO: Not threadsafe, or persistent
-func (account *Account) IncrementId() {
-	// TODO: Needs to be done correctly
-	account.NextId += 1
-}
+var current *Authentication
 
 func NewAuthentication() *Authentication {
```

nit: Naming, to me NewAuthentication implies that this is a builder (in java terms a constructor) for an object typ, but the latest changes appear to return more like a Singleton. The way the golang standard library appears to deal with this, using DNS as an example, is by having DefaultResolver. So in this case the function could be called something more like DefaultAuthentication to return the default implementation that is locally relevant.

---

I kinda like prefacing it with Default, but isn't that a bit misleading if the struct is initialized from persistence? Ironically, it feels like I should call it OldAuthentication :-) (note: I don't really like calling the struct Authentication either, it's misnamed as well.)